src/include/distributed/local_executor.h

```diff
@@ -17,8 +17,14 @@
 extern bool EnableLocalExecution;
 extern bool LogLocalCommands;
 
-extern bool TransactionAccessedLocalPlacement;
-extern bool TransactionConnectedToLocalGroup;
+typedef enum LocalExecutionStatus
+{
+	LOCAL_EXECUTION_REQUIRED,
+	LOCAL_EXECUTION_OPTIONAL,
+	LOCAL_EXECUTION_DISABLED
+} LocalExecutionStatus;
```

Comment:
State could be a better word here, but in postgres State is used more broadly so not sure what we should choose here